packages/@lwc/engine/src/framework/api.ts

```diff
@@ -412,6 +424,57 @@ export function c(
     return vnode;
 }
 
+/**
+ * web component compatible vnode: this method is equivalent to c()
+ * but it has an extra check to determine if the constructor is a
+ * web component or LWC, and depending on that check, we build a
+ * VElement or a VCustomElement virtual node.
+ *
+ * Note: The reason for this specialized method is perf. We don't want
+ *       to make the instanceof check for every VCustomElement that we
+ *       create, that's too costly, apparently.
+ *
+ * Additionally, if a regular VElement is created, it will add an extra
+ * check to validate that at the end of the diffing algo, the element
+ * is in fact driven by a registered custom element, otherwise throw.
+ *
+ **/
+export function wc(
```

Comment:
@diervo here is the new api, now we need to figure how to enable this in compiler.

Reply:
the alternative to be considered here is that for open source, we always use wc() unless a config is provided into compiler, which then uses c() if not wc is allowed.